### PR TITLE
Implement .floor, .ceil, .trunc for the in-memory Decimal column

### PR DESCRIPTION
**Pull request by GregoryTravis**
### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.

### CODE REVIEW

**Review by jdunkerley — APPROVED** [review]

File: distribution/lib/Standard/Table/0.0.0-dev/src/Column.enso [hunk 1]
```diff
@@ -954,7 +958,11 @@ type Column
                 new_name = naming_helper.function_name "ceil" [self]
                 self.rename new_name
             False ->
-                apply_unary_operation self UnaryRoundOperation.CEIL_INSTANCE
+                case self.inferred_precise_value_type.is_decimal of
+                    True ->
+                        apply_unary_operation self UnaryDecimalRoundOperation.CEIL_INSTANCE
+                    False ->
+                        apply_unary_operation self UnaryRoundOperation.CEIL_INSTANCE
```
Comment (radeusgd) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Column.enso:965]: Is there any reason why we cannot merge `UnaryDecimalRoundOperation` with `UnaryRoundOperation`?

I thought the whole idea behind the (relatively new) `UnaryXyzOperation` architecture was to decouple operations from storages.

Comment (radeusgd) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Column.enso:965]: But maybe I misunderstood the `UnaryXyzOperation` approach. If you think that splitting them is better, then I think it is okay to keep this as-is.

**Review by radeusgd — APPROVED** [review]
Looks good to me, as written I'm wondering how we should approach our unary operations (as we probably will be moving binary ops to similar architecture some time in the future).

IMO there should be a single `UnaryRoundOperation`. My motivation is - if we will migrate binary operations to this architecture, we'll prefer to have a single `BinaryAddOperation` that is capable of handling all kinds of parameter combinations (`int64 + float`, `float + int32` etc. etc.) - if we had a separate class for each combination it would be unmanageable I think.

On the other hand, I do appreciate the 'decoupling' that keeping the separate unary implementations distinct. They are a slightly different case since they do not suffer from the combination of parameters problem. 

What do you think?

File: distribution/lib/Standard/Table/0.0.0-dev/src/Column.enso (hunk 1 above)
Comment (GregoryTravis) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Column.enso:965]: I think they should be merged, but I was trying to preserve the existing structure until I was clear about the new plan. I have another task #10904 that will require merging these two, so if it's okay I'll merge them in that PR (which will happen as soon as this and #10344 are done).

**Pull request merged**